addons/mail/static/src/core/web/discuss_client_action.js

```diff
@@ -53,15 +53,18 @@ export class DiscussClientAction extends Component {
             // legacy format (sent in old emails, shared links, ...)
             model = "discuss.channel";
         }
-        const activeThread = this.store.Thread.get({ model, id });
-        if (!activeThread || activeThread.notEq(this.store.discuss.thread)) {
-            const thread =
-                this.store.Thread.get({ model, id }) ??
-                (await this.threadService.fetchChannel(parseInt(id)));
-            if (!thread.is_pinned) {
-                await this.threadService.pin(thread);
+        let activeThread = this.store.Thread.get({ model, id });
+        if (activeThread && activeThread.eq(this.store.discuss.thread)) {
+            return;
+        }
+        if (!activeThread?.type && model === "discuss.channel") {
+            activeThread = await this.threadService.fetchChannel(parseInt(id));
+            if (!activeThread.is_pinned && !activeThread.readonly) {
+                await this.threadService.pin(activeThread);
```

Comment:
Thread should probably never be pinned here, I'm not sure what it is trying to fix, possibly nothing, but if it does, it doesn't look like the right place to do it.

Reply:
My first attempt was to adapt `pin()` to execute when all the conditons are met. After reading [your comment](https://github.com/odoo/odoo/pull/137623#discussion_r1371696777) I thought you expected me to dive deeper in the call stack to only invoke this method if the conditions allow it. This is the code that "automatically" pin channels so it appeared to me as the right place to add this new `readonly` conditon.